### PR TITLE
Handle Multiple Accounts

### PR DESCRIPTION
**Pull request by InvestYGator**
Try to add tracking of how many broadcasts we have done per account, per block.  If we hit the limit, leave them in the queue.

### CODE REVIEW

File: hive-interface.js [hunk 1]
```diff
@@ -214,13 +216,21 @@ class Hive {
 
 	// Left for backwards compatibility
 	async customJsonNoQueue(id, json, account, key, use_active) {
+		if(this.accounts_used.has(account)) {
+			let count = this.accounts_used.get(account);
+			if(count => 4) 
+				// If we are at the limit, queue instead
+				return this.custom_json(id, json, account, key, use_active);
+			else
```
Comment (TravisForkner) [inline hive-interface.js]: I think you need to move this down or you will never set the account.

File: hive-interface.js [hunk 2]
```diff
@@ -418,11 +428,53 @@ class Hive {
...
+			for (const key of this.accounts_used.keys()) {
+			this.accounts_used.set(key, 0);
+			}
+			this.clear_ctr = 0;
+		}
+		// If the queue is empty, exit here
+		if(this.tx_queue.length <= 0) return;		
+		let exit = false;
+		let next_account = this.tx_queue[0].data.required_auths.length > 0 ? this.tx_queue[0].data.required_auths[0] : this.tx_queue[0].data.required_posting_auths[0];
+		if(this.accounts_used.has(next_account)) {
```
Comment (TravisForkner) [inline hive-interface.js]: since we reuse this functionality in multiple places you may want to create method for it.

**Review by TravisForkner — CHANGES_REQUESTED** [review]

File: hive-interface.js (hunk 1 above)
Comment (InvestYGator) [reply hive-interface.js]: Ah, yea, need to add the line handling new accounts at the bottom of that code.

File: hive-interface.js [hunk 3]
```diff
@@ -413,16 +418,53 @@ class Hive {
 		});
 	}
 
+	checkAccountUsageLimit(account) {
+		if(this.accounts_used.has(account)) {
+			let count = this.accounts_used.get(account);
+			if(count => 4)
```
Comment (TravisForkner) [inline hive-interface.js]: this check is backwards

**Review by TravisForkner — CHANGES_REQUESTED** [review]

**Review by TravisForkner — APPROVED** [review]

**Review by farpetrad-sl — APPROVED** [review]

**Pull request merged**